lib/rubygems/specification.rb

```diff
@@ -2711,11 +2711,18 @@ def validate packaging = true
         raise Gem::InvalidSpecificationException,
           "each license must be 64 characters or less"
       end
+
+      if !Gem::SPDX::IDENTIFIERS.include?(license)
+        warning <<-warning % license
+WARNING: license value '%s' is invalid.  Use a license identifier from:
+http://spdx.org/licenses
+        warning
```

Comment:
@bf4, I'll have to plead ignorance here. Is it clear the template should be frozen, or are there trade-offs (perf, resiliency to runtime modification) to be assessed?